src/Clash/Cores/Sgmii/PcsTransmit.hs

```diff
@@ -1,18 +1,61 @@
 {- |
-  Copyright   :  (C) 2024, QBayLogic B.V.
+  Copyright   :  (C) 2024-2025, QBayLogic B.V.
   License     :  BSD2 (see the file LICENSE)
   Maintainer  :  QBayLogic B.V. <devops@qbaylogic.com>
 
   Top level module for the PCS transmit block, that combines the processes
   that are defined in the two submodules @CodeGroup@ and @OrderedSet@.
 -}
-module Clash.Cores.Sgmii.PcsTransmit (pcsTransmit) where
+module Clash.Cores.Sgmii.PcsTransmit
+  ( InputDelayState
+  , pcsTransmit
+  , inputDelayT
+  )
+where
 
 import Clash.Cores.Sgmii.Common
 import Clash.Cores.Sgmii.PcsTransmit.CodeGroup
 import Clash.Cores.Sgmii.PcsTransmit.OrderedSet
 import Clash.Prelude
 
+-- | Tuple that contains a queue containing the most recent incoming octets and
+--   an 'Index' that points to the current octet to be transmitted.
+type InputDelayState n = (Index n, Vec n (Bool, Bool, BitVector 8))
+
+-- | Transition function for the input queue that outputs the correct octet when
+--   the system is ready. Note that, when the 'Index' that points to the next
+--   value to be outputted starts going down, it should act one time instance
+--   earlier than when it goes up.
```

Comment:
Could you shortly mention the justification that the statemachine has states where it can not consume the input, but the GMII protocol does not feature a backpressure mechanism.